python/tvm/contrib/pipeline_executor.py

```diff
@@ -0,0 +1,533 @@
+                owner_name = self.io_owner.name
+
+            return owner_name, self.name
+
+        def get_owner_idx(self):
+            """Return owner idex if owner is ModuleWrapper, if not return 0"""
+            if isinstance(self.io_owner, PipelineConfig.ModuleWrapper):
+                return self.io_owner.idx
+
+            # If not ModuleWrapper then owner is PipelineConfig, return 0
+            # to identify this is global interface
+            return 0
+
+        def is_global_interface(self):
+            """Check if this interface is global interface."""
+            return not isinstance(self.io_owner, PipelineConfig.ModuleWrapper)
+
+        def __repr__(self):
+            """Get all binding(input data), exepect like |data_0: mod1:data_0"""
+            ret = "  |{}: ".format(self.name)
+            for binding in self.bindings:
+                mname, dname = binding.get_name()
+                ret += "{0}:{1} ".format(mname, dname)
+            return ret
+
+        def check_dag_acyclic(self, start, inputs):
+            """Check if the DAG that current binding stay is acircle"""
+            for binding in inputs.values():
+                if start == binding.io_owner:
+                    return False
+                for p in binding.parents:
+                    if not self.check_dag_acyclic(start, p.io_owner.input_bindings.bindings):
+                        return False
+
+            return True
+
+        def connect(self, binding):
+            """
+            # Check if the bindendency setting correct.
+            # correct connection are following
+            # 1. global input to module input
+            # 2. module output to global output
+            # 3. module output to moudle input
+            """
+            if self.io_owner == binding.io_owner:
+                raise RuntimeError(f"can not set self as binding.")
+
+            if not self.is_global_interface() and self.io_type == "input":
+                raise RuntimeError(f"Module only can start binding from output!")
+
+            if (
+                not self.is_global_interface()
+                and not binding.is_global_interface()
+                and binding.io_type == "output"
+            ):
+                raise RuntimeError(f"Module output can not binding with module output!")
+
+            if (
+                not self.is_global_interface()
+                and binding.is_global_interface()
+                and binding.io_type == "input"
+            ):
+                raise RuntimeError(f"Module output can not binding with global input!")
+
+            if self.is_global_interface() and self.io_type != "input":
+                raise RuntimeError(f"Global only can start binding from input!")
+
+            if self.is_global_interface() and binding.io_type != "input":
+                raise RuntimeError(f"Global input only can set binding with module input.")
+
```

Comment:
Please fix and polish ALL error messages in `RuntimeError` above, from L238

Reply:
fixed.